fastparquet/writer.py

```diff
@@ -1486,3 +1487,52 @@ def sort_key(row_group) -> int:
 def write_thrift(f, obj):
     # TODO inline this
     return f.write(obj.to_bytes())
+
+def update_file_metadata(path: str, custom_metadata: dict,
+                         is_metadata: bool = None):
+    """Update metadata in file without rewriting data portion if a data file.
+
+    Update strategy depends if key found in new custom metadata is also found
+    in already existing custom metadata within thrift object, as well as its
+    value.
+        
+      - If not found in existing, it is added.
+      - If found in existing, it is updated.
+      - If its value is `None`, it is not added, and if found in existing,
+        it is removed from existing.
+
+    Parameters
+    ----------
+    path : str
```

Comment:
This is a *local* path, this does not work for any other filesystem.

Reply:
Ok, I added the information, and also a note to the docstring:

```python
    path : str
        Local path to file.

[...]

    Notes
    -----
    This method does not work for remote files.
```